### DISCUSSION TITLE
Reduce library size

### DISCUSSION THREAD

**Discussion by reg31** [openssl/openssl#25044] [Q&A]
Hi,
Is there a compile flag to reduce the size of the openssl binaries?

**Comment by mattcaswell** [comment]
There are various flags which might have the effect of reducing the size by omitting support for various things.

You might try:

````
./Configure  no-argon2 no-aria no-async no-bf no-blake2 no-camellia no-cast no-cmp no-cms no-comp no-deprecated no-des no-dgram no-dh no-dsa no-ec2m no-engine no-gost no-http no-idea no-legacy no-md4 no-mdc2 no-multiblock no-nextprotoneg no-ocb no-ocsp no-quic no-rc2 no-rc4 no-rmd160 no-scrypt no-seed no-siphash no-siv no-sm2 no-sm3 no-sm4 no-srp no-srtp no-ts no-whirlpool -Os
````
Obviously don't turn off things you might want to actually use.

**Comment by reg31** [comment]
Thank you

**Comment by paulidale** [comment]
Some additional savings can also be had by specifying `-DOPENSSL_SMALL_FOOTPRINT` on the configure line.  This trades space for time in various places.
